source/Fleet.h

```diff
@@ -73,6 +75,17 @@ class Fleet {
 
 	int64_t Strength() const;
 
+	// Updates any data that relies on conditions
+	void UpdateConditions();
```

Comment:
One of the reasons for switching from "Auto Conditions generated on every switch" to "Auto Conditions generated on demand" was to avoid the need to manually call update functions (which can lead to outdated information being used if the function is not called often enough).
Is it possible to do something similar here?

Reply:
Yes. I was going to use a notification system, that would send signals to objects when critical events happened, like a mission action completing, but that would add another ~500 lines to the PR.

As for the obvious alternative, having Conditions just magically update, has been suggested already in this PR twice, and it would be astoundingly bad. Take this example:

1. Create Conditions connected to a ConditionsStore
2. Use those Conditions in a WeightedList. It'll correct the weights and generate the total.
3. Some missions complete, changing the values of the conditions that WeightedList was using.
4. The WeightedList weights don't add up to the total, and WeightedList had no way to know that would happen.

Reply:
Oh, and you already can update the conditions on demand. You do that by calling UpdateConditions just before accessing it each Condition. The code examples in the FAQ shows how that can be done.

Anything with a WeightedList *must* have the recursive UpdateConditions functionality. This is also explained in the FAQ.